IBPSA/Fluid/Chillers/BaseClasses/InnerCycle_Chiller.mo

```diff
@@ -0,0 +1,209 @@
+within IBPSA.Fluid.Chillers.BaseClasses;
+model InnerCycle_Chiller "Blackbox model of refrigerant cycle of a chiller"
+  extends IBPSA.Fluid.HeatPumps.BaseClasses.PartialInnerCycle;
+  // Setting all values to zero avoids errors when checking this model.
+  // The values are correctly propagated by the heat pump / chiller model anyway
+  replaceable model BlaBoxChiCooling =
+      IBPSA.Fluid.Chillers.BlackBoxData.BaseClasses.PartialChillerBlackBox (
+       QUse_flow_nominal=0,
+       QUseBlackBox_flow_nominal=0,
+       scalingFactor=0,
+       TCon_nominal=0,
+       TEva_nominal=0,
+       dTCon_nominal=0,
+       dTEva_nominal=0,
+       primaryOperation=true,
+       mCon_flow_nominal=0,
+       mEva_flow_nominal=0,
+       y_nominal=0)
+    constrainedby
+    IBPSA.Fluid.Chillers.BlackBoxData.BaseClasses.PartialChillerBlackBox
+    "Replaceable model for performance data of a chiller in main operation mode"
+    annotation (choicesAllMatching=true);
```

Comment:
Adding `partial` does not work for me, the error still occurs. Does it work for you in another version of Dymola? Which are you using?

Reply:
I have Dymola 2023 (64-bit).
As I test this with your updated code (commit [07606b0](https://github.com/ibpsa/modelica-ibpsa/pull/1628/commits/07606b0281765ddb86aea676eda50efad24fd4f0)), for `IBPSA.Fluid.Chillers.BaseClasses.BlackBoxVapourCompressionCycle`, without the `partial` keyword I have the following error message upon checking the model:
```
Component type specifier BlackBoxChillerCooling is equal to a partial type
It must be redeclared before being used.
```
With the `partial` keyword at the beginning, I can delete the whole bracket of modifications without triggering any 
error message.

Reply:
Do you mean like this?
```
  replaceable partial model RefrigerantCycleHeatPumpHeating =
      IBPSA.Fluid.HeatPumps.RefrigerantCycleModels.BaseClasses.PartialHeatPumpRefrigerantCycle
     constrainedby 
    IBPSA.Fluid.HeatPumps.RefrigerantCycleModels.BaseClasses.PartialHeatPumpRefrigerantCycle
    "Replaceable model for refrigerant cycle of a heat pump in main operation mode"
    annotation (choicesAllMatching=true);
```
My output:
```
Check of IBPSA.Fluid.HeatPumps.BaseClasses.HeatPumpRefrigerantCycle:

Component type specifier RefrigerantCycleHeatPumpHeating is a partial type
It must be redeclared before being used.
Modelica Text: line 22
Component context: refCycHeaPumHea
Component declared as RefrigerantCycleHeatPumpHeating refCycHeaPumHea in IBPSA.Fluid.HeatPumps.BaseClasses.HeatPumpRefrigerantCycle

Component type specifier RefrigerantCycleHeatPumpCooling is a partial type
It must be redeclared before being used.
Modelica Text: line 25
Component context: refCycHeaPumCoo
Component declared as RefrigerantCycleHeatPumpCooling refCycHeaPumCoo in IBPSA.Fluid.HeatPumps.BaseClasses.HeatPumpRefrigerantCycle

The model has 65 scalar unknowns and 61 scalar equations.

The following parameters don't have any value:
refCycHeaPumCoo.PEle_nominal
refCycHeaPumHea.dTCon_nominal
refCycHeaPumHea.dTEva_nominal
refCycHeaPumHea.mCon_flow_nominal
refCycHeaPumHea.mEva_flow_nominal
refCycHeaPumHea.PEle_nominal
refCycHeaPumHea.QUse_flow_nominal
refCycHeaPumHea.QUseNoSca_flow_nominal
refCycHeaPumHea.y_nominal

DAE with 67 unknown scalars and 63 scalar equations.

ERRORS have been issued.
```
I am using Dymola Version 2023 (64-bit), 2022-04-13.

Reply:
No I mean in `IBPSA.Fluid.Chillers.BaseClasses.ChillerRefrigerantCycle`, declare the class as
`
partial model ChillerRefrigerantCycle "Refrigerant cycle model of a chiller"
`
at the first (second if counting the `within` statement) line.

Reply:
This won't work, as we use the model in the `ModularReversible` models directly. It would create an error, that the `ChillerRefrigerantCycle` is partial and has to be redeclared.